module/input/SensorFilter/src/SensorFilter.cpp

```diff
@@ -564,146 +571,141 @@ namespace module::input {
                                              MeasurementType::GYROSCOPE());
 
                         // Calculate accelerometer noise factor
-                        Eigen::Matrix3d acc_noise =
+                        const Eigen::Matrix3d acc_noise =
                             config.motionFilter.noise.measurement.accelerometer
+                            // Add noise which is proportional to the square of how much we are moving, minus gravity
+                            // This means that the more we're accelerating, the noisier we think the measurements are
                             + ((sensors->accelerometer.norm() - std::abs(G))
                                * (sensors->accelerometer.norm() - std::abs(G)))
                                   * config.motionFilter.noise.measurement.accelerometerMagnitude;
 
                         // Accelerometer measurement update
                         motionFilter.measure(sensors->accelerometer, acc_noise, MeasurementType::ACCELEROMETER());
-
-                        for (auto& side : {BodySide::LEFT, BodySide::RIGHT}) {
-                            bool foot_down      = sensors->feet[side].down;
-                            bool prev_foot_down = previous_foot_down[side];
-                            Eigen::Affine3d Htf(
+                        // This loop calculates the Hwf transform for feet if they have just hit the ground. If they
+                        // have not just hit the ground, it uses the previous Hwf value. This assumes that once the foot
```

Comment:
i.e, we assume our foot cant slip (make our assumptions explicit)

Reply:
I have added an extra clause to make it more explicit :+1: